test/parallel-benchmark/mzcompose.py

```diff
@@ -0,0 +1,1190 @@
+    )
+
+    failures: list[TestFailureDetails] = []
+
+    if args.mz_url:
+        with c.override(
+            Testdrive(
+                no_reset=True,
+                materialize_url=args.mz_url,
+                seed=1,
+                no_consistency_checks=True,
+            )
+        ):
+            target = parse_pg_conn_string(args.mz_url)
+            c.up("testdrive", persistent=True)
+            conn_infos = {"materialized": target}
+            conn = target.connect()
+            with conn.cursor() as cur:
+                mz_version = cur.execute("SELECT mz_version()")
+                mz_version = cur.fetchall()[0][0]
+            conn.close()
+            mz_string = f"{mz_version} ({target.host})"
+    else:
+        with c.override(
+            Materialized(
+                default_size=args.size,
+                soft_assertions=False,
+                external_cockroach=True,
+                external_minio=True,
+                sanity_restart=False,
+                additional_system_parameter_defaults={
+                    "enable_statement_lifecycle_logging": "false",
+                    "statement_logging_default_sample_rate": "0",
+                    "statement_logging_max_sample_rate": "0",
+                },
+            )
+        ):
+            c.up(*service_names)
+            c.up("testdrive", persistent=True)
+            c.sql(
+                "ALTER SYSTEM SET max_connections = 1000000",
+                user="mz_system",
+                port=6877,
+            )
+
+            mz_version = c.query_mz_version()
+            mz_string = f"{mz_version} (docker)"
+            conn_infos = {
+                "materialized": PgConnInfo(
+                    user="materialize",
+                    database="materialize",
+                    host="127.0.0.1",
+                    port=c.default_port("materialized"),
+                ),
+                "mz_system": PgConnInfo(
+                    user="mz_system",
+                    database="materialize",
+                    host="127.0.0.1",
+                    port=c.port("materialized", 6877),
+                ),
+                "postgres": PgConnInfo(
+                    user="postgres",
+                    password="postgres",
+                    database="postgres",
+                    host="127.0.0.1",
+                    port=c.default_port("postgres"),
+                ),
+            }
+
+    c.silent = True
+    for scenario_class in scenarios:
+        scenario_name = scenario_class.__name__
+        print(f"--- Running scenario {scenario_name}")
+        measurements = collections.defaultdict(list)
+        stats: dict[str, Statistics] = {}
+        scenario = scenario_class(c, conn_infos)
+        scenario.setup(c, conn_infos)
+        start_time = time.time()
+        Path(MZ_ROOT / "plots").mkdir(parents=True, exist_ok=True)
+        try:
+            # Don't let the garbage collector interfere with our measurements
```

Comment:
I guess we need this for other benchmarks (feature benchmark, scalability framework) as well?